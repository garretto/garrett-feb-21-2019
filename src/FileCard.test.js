import React from 'react'
import ReactDOM from 'react-dom'
import FileCard from './FileCard.js'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<FileCard />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <FileCard
        name="FileName.jpg"
        bytes="1000000"
        onDelete={() => console.log('delete')}
        deleting={false}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders deleting correctly', () => {
  const tree = renderer
    .create(
      <FileCard
        name="FileName.jpg"
        bytes="1000000"
        onDelete={() => console.log('delete')}
        deleting={true}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
