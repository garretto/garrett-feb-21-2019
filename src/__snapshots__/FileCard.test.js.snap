// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="FileCard"
>
  <h2>
    FileName.jpg
  </h2>
  <div
    className="file-details"
  >
    <div
      className="file-size"
    >
      1mb
    </div>
    <button
      className="button is-danger is-outlined "
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "width": 80,
        }
      }
    >
      delete
    </button>
  </div>
</div>
`;

exports[`renders deleting correctly 1`] = `
<div
  className="FileCard"
>
  <h2>
    FileName.jpg
  </h2>
  <div
    className="file-details"
  >
    <div
      className="file-size"
    >
      1mb
    </div>
    <button
      className="button is-danger is-outlined is-loading"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "width": 80,
        }
      }
    >
      delete
    </button>
  </div>
</div>
`;
